releases/changelog.md

```diff
@@ -0,0 +1,73 @@
+# Changelog
+
+In our repositories we keep `CHANGELOG.md` files describing changes between
+releases of the project. In general we follow [keep
+a changelog][keepachangelog] guidelines.
+
+**The changelog should be able to answer two questions:**
+
+- Is there a reason to upgrade to the new version?
+- What breaking changes were introduced? How the migration to the new
+  version looks like?
+
+**Those questions should be answered from the perspective of the user of the
+repository.** The user can be something else for different types of codebases:
+
+- For CLI program: a person or entity calling the binary.
+- For managed app: a person or entity installing the application.
+- For library: another codebase using the library.
+- For operator: a person or entity creating the CR consumed by this operator.
+
+## Examples of reasons to upgrade
+
+- Bug fix.
+- Security fix.
+- New functionality.
+- Improved existing functionality.
+
+## Examples of breaking changes
+
+For CLI program:
+
+- Renamed command or flags.
+- Changed behaviour of a command.
+
+For managed app:
+
+- Renamed value in chart's values.yaml file.
+
+For library:
+
+- Changed API function name or arguments.
+- Changed behaviour of API function.
+
+For operator:
+
+- Changed defaults for CR empty fields.
+- Changed the desired state of the system for the same CR. E.g. not public but
+  internal load-balancer is created for the same CR.
+
+## Examples of noise entries
+
+We should try to not mechanically write in changelog everything that is changed
```

Comment:
I hadn't picked up the downsides of noise changelog entries. So IMO these docs help a lot.

But we added this to the PR checklist template which may be contributing.

- [ ] Update changelog in CHANGELOG.md.

Should it be this?

- [ ]  Update changelog in CHANGELOG.md (if needed).

Reply:
That's a good call. I also wanted to make it a link to this page like this:

> - [ ] [Update changelog in CHANGELOG.md (if needed)](https://github.com/giantswarm/fmt/blob/34f6186bca7fed497a4c3eb676edb79127bde7a1/releases/changelog.md).

Reply:
Nice +1 on linking to these docs.